cdap-ui/server/config/parser.js

```diff
@@ -28,34 +29,61 @@ var promise = require('q'),
     log4js = require('log4js'),
     cache = {},
     path,
-    buffer = '';
+    buffer = '',
+    request = require('request');
 
 var log = log4js.getLogger('default');
 
 function extractUISettings() {
   try {
     if (require.resolve('./ui-settings.json')) {
-      const uiSettings = require('./ui-settings.json') || {};
-      const uiThemeName = uiSettings['ui.theme'] || 'default';
-      const uiThemePath = `./themes/${uiThemeName}.json`;
-      let uiThemeConfig = {};
-      try {
-        if (require.resolve(uiThemePath)) {
-          uiThemeConfig = require(uiThemePath);
-          log.info(`UI using ${uiThemeName} theme`);
-        }
-      } catch (err) {
-        // The error can either be file doesn't exist, or file contains invalid json
-        log.info(err.toString());
-      }
-      return {uiSettings, uiThemeConfig};
+      return require('./ui-settings.json') || {};
     }
-  } catch(e) {
+  } catch (e) {
     log.info('Unable to find UI settings json file.');
     return {};
   }
 }
 
+/*
+ *  Extracts contents of theme object in the specified theme file
+ *  @returns {promise}
+ */
+
+function extractUITheme() {
+  const uiThemePropertyName = 'ui.theme.file';
+
+  return extractConfig('cdap')
+    .then(function (cdapConfig) {
+      const deferred = promise.defer();
+
+      getCdapSiteContents(cdapConfig, function (err, response, body) {
```

Comment:
Can we try this using `async ... await`? Something like,

```Javascript

async function extractUITheme () {
  const uiThemePropertyName = 'ui.theme.file';
  const DEFAULT_CONFIG = {};
  let cdapConfig = await extractConfig('cdap');
  let cdapSiteContents = await getCDAPSiteContents(cdapConfig);
  if (cdapSiteContents.error) {
    log.info('Unable to get CDAP config');
    return DEFAULT_CONFIG;
  }
  try {
    cdapSiteContents = JSON.parse(body);
    const uiThemeProperty = cdapSiteContents.find(property => property.name === uiThemePropertyName);
    if (!uiThemeProperty) {
      log.info(`Unable to find ${uiThemePropertyName} property`);
      return DEFAULT_CONFIG;
    }
    const uiThemeConfig = getUIThemeConfig(uiThemeProperty);
    return uiThemeConfig;
  } catch(e) {
    log.info('Unable to parse CDAP config');
    return DEFAULT_CONFIG;
  }
}
```
Sorry should have communicated right the first time. The callbacks become hard to read when we nest things.

Reply:
async await was available from node 7 I believe. Do check if its available in 8